woocommerce/woocommerce
# Revert PR #17895

### Changes proposed in this Pull Request:

This PR reverts #17895 which was introduced in WC 3.5 to use post_author instead of `_customer_user` post meta to store customer ID for orders. After talking to @claudiulodro, we decided to revert the changes for WC 3.5.1 and reconsider #17895 for 3.6. This decision was made due to the bugs that were introduced by #17895 and that we didn't catch before the release (#21656, #21687 and #21737). All the issues seem fixable, but we are opting to err on the side of caution instead of rushing to create fixes that could introduce more problems. Reverting is the safest and fastest option that we have now.

#17895 copied data from `_customer_user` postmeta to the field `post_autor` of the table `wp_posts` and changed WooCommerce to populate both `customer_user` and `post_author` when a order is created or updated. So, this PR simply revert the code changes to make WooCommerce get the customer ID for orders in the old way. For now, we are not touching the data that was copied to `post_author` and this shouldn't be a problem as WooCommerce doesn't use this field for orders.

It was not possible to revert #17895 automatically, so I had to revert each commit manually. I decided to keep the changes introduced in `includes/class-wc-background-updater.php` (https://github.com/woocommerce/woocommerce/pull/17895/files#diff-3809cc9b6218b1f645071ed0351b2edc) as they could be useful for future database update routines.

Closes #21656, #21687, #21737 and #21741.
Supersedes #21673 and #21695

### How to test the changes in this Pull Request:

1. Check that code changes were properly reverted
2. Check that the issue in #21737 was fixed
3. Test that the functionality altered in this PR still works: customer reports, filter orders by customer in the admin interface, filter orders by customer in the REST API, saving customer ID when placing an order or manually created it via wp-admin, saving customer ID when updating an order, get customer last order (https://github.com/woocommerce/woocommerce/compare/revert/17895?expand=1#diff-d6dba1e5e24baa6d5d15671305dd1994R325), `wc_reset_order_customer_id_on_deleted_user()` and `wc_customer_bought_product()`.
4. Make sure that orders that were created after the WC 3.5 have the correct customer after updating to 3.5.1.

### Changelog entry

> Revert PR to use post_author instead of the post meta '_customer_user' to store customer ID for orders
